ndcube/tests/test_ndcube.py

```diff
@@ -128,30 +129,79 @@ def test_slicing_split_celestial(ndc, item):
                                                               [False, True]], dtype=bool))
 
 
+def test_axis_world_coords_wave_ec(ndcube_3d_l_ln_lt_ectime):
+    cube = ndcube_3d_l_ln_lt_ectime
+
+    coords = cube.axis_world_coords('em.wl')
+    assert u.allclose(coords, [1.02e-09, 1.04e-09, 1.06e-09, 1.08e-09, 1.10e-09,
+                               1.12e-09, 1.14e-09, 1.16e-09, 1.18e-09, 1.20e-09] * u.m)
+
+    coords = cube.axis_world_coords()
+    assert len(coords) == 2
+
+    coords = cube.axis_world_coords(wcs=cube.combined_wcs)
+    assert len(coords) == 3
+
+    coords = cube.axis_world_coords(wcs=cube.extra_coords)
+    assert len(coords) == 1
+    assert isinstance(coords[0], Time)
+    assert coords[0].shape == (5,)
+
+    coords = cube.axis_world_coords_values(wcs=cube.extra_coords)
+    assert len(coords) == 1
+    assert isinstance(coords[0], u.Quantity)
+    assert coords[0].shape == (5,)
+
+
+def test_axis_world_coords_complex_ec(ndcube_4d_ln_lt_l_t):
+    cube = ndcube_4d_ln_lt_l_t
+    ec_shape = cube.data.shape[1:3]
+    data = np.arange(np.product(ec_shape)).reshape(ec_shape) * u.m / u.s
+
+    # The lookup table has to be in world order so transpose it.
+    cube.extra_coords.add_coordinate('velocity', (1, 2), data.T)
```

Comment:
My earlier comment [in chat](https://matrix.to/#/!TWSJXVpyflnZOzmSQz:matrix.org/$1605789849477PNHzH:cadair.com?via=matrix.org&via=openastronomy.org&via=cadair.com) is related to this. When passing the world lookup table here, it has to be in world order not array order.

Reply:
I have opened #342 to track this